src/xenia/ui/imgui_drawer.cc

```diff
@@ -218,6 +192,58 @@ std::optional<ImGuiKey> ImGuiDrawer::VirtualKeyToImGuiKey(VirtualKey vkey) {
   }
 }
 
+void ImGuiDrawer::InitializeFonts() {
+  auto& io = ImGui::GetIO();
+
+  const float default_font_size = 12.0f;
+  // TODO(gibbed): disable imgui.ini saving for now,
+  // imgui assumes paths are char* so we can't throw a good path at it on
+  // Windows.
+  io.IniFilename = nullptr;
+
+  // Setup the font glyphs.
+  ImFontConfig font_config;
+  font_config.OversampleH = font_config.OversampleV = 2;
+  font_config.PixelSnapH = true;
+
+  if (!cvars::custom_font_path.empty() &&
+      std::filesystem::exists(cvars::custom_font_path)) {
+    const std::string font_path = xe::path_to_utf8(cvars::custom_font_path);
+    ImFont* font = io.Fonts->AddFontFromFileTTF(
+        font_path.c_str(), default_font_size, &font_config,
+        io.Fonts->GetGlyphRangesDefault());
+
+    io.Fonts->Build();
+    // Something went wrong while loading custom font. Probably corrupted.
+    if (!font->IsLoaded()) {
+      XELOGE("Failed to load custom font: {}", font_path);
+      io.Fonts->Clear();
+    }
+  }
+
+  if (io.Fonts->Fonts.empty()) {
+    io.Fonts->AddFontFromMemoryCompressedBase85TTF(
+        kProggyTinyCompressedDataBase85, default_font_size, &font_config,
+        io.Fonts->GetGlyphRangesDefault());
+  }
+
+  // TODO(benvanik): jp font on other platforms?
+  // https://github.com/Koruri/kibitaki looks really good, but is 1.5MiB.
+  const char* jp_font_path = "C:\\Windows\\Fonts\\msgothic.ttc";
```

Comment:
This should use the `windir` environment variable.

Reply:
Done, but in a way less prone to errors